### PR TITLE
[TiKV] Introduce `raft-engine.prefill-for-recycle` configuration.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add a new configuration `raft-engine.prefill-for-recycle` in TiKV for supporting to make log recycling effective when starting in cold state.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/13339
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -1809,6 +1809,15 @@ Configuration items related to Raft Engine.
 + Determines whether to recycle stale log files in Raft Engine. When it is enabled, logically purged log files will be reserved for recycling. This reduces the long tail latency on write workloads.
 + Default value: `false`
 
+### `prefill-for-recycle` <span class="version-mark">New in 7.0.0</span>
+
+> **Note:**
+>
+> This configuration item is only available when [`enable-log-recycle`](#enable-log-recycle-new-in-v630) >= 2.
```
Comment (ran-huang) [inline tikv-configuration-file.md:1816]:
```suggestion
> This configuration item only takes effect when [`enable-log-recycle`](#enable-log-recycle-new-in-v630) >= 2.
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -1809,6 +1809,15 @@ Configuration items related to Raft Engine.
...
+
++ Whether to generate empty log files for log recycling in Raft Engine. When it is enabled, Raft Engine will automatically fill a batch of empty log files for log recycling during initialization, making log recycling effective immediately after initialization.
```
Comment (ran-huang) [inline tikv-configuration-file.md:1818]:
```suggestion
+ Determines whether to generate empty log files for log recycling in Raft Engine. When it is enabled, Raft Engine will automatically fill a batch of empty log files for log recycling during initialization, making log recycling effective immediately after initialization.
```

**Review by ran-huang — COMMENTED** [review]
Rest LGTM

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1809,6 +1809,15 @@ Configuration items related to Raft Engine.
 + Determines whether to recycle stale log files in Raft Engine. When it is enabled, logically purged log files will be reserved for recycling. This reduces the long tail latency on write workloads.
 + Default value: `false`
 
+### `prefill-for-recycle` <span class="version-mark">New in 7.0.0</span>
+
+> **Note:**
+>
+> This configuration item only takes effect when [`enable-log-recycle`](#enable-log-recycle-new-in-v630) >= 2.
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1816]:
```suggestion
> This configuration item only takes effect when [`enable-log-recycle`](#enable-log-recycle-new-in-v630) is set to `true`.
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1809,6 +1809,15 @@ Configuration items related to Raft Engine.
 + Determines whether to recycle stale log files in Raft Engine. When it is enabled, logically purged log files will be reserved for recycling. This reduces the long tail latency on write workloads.
 + Default value: `false`
 
+### `prefill-for-recycle` <span class="version-mark">New in 7.0.0</span>
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1812]:
```suggestion
### `prefill-for-recycle` <span class="version-mark">New in v7.0.0</span>
```

**Review by Oreoxmt — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**